### PR TITLE
Implement Redux store and display books

### PR DESCRIPTION
**Pull request by mershark**
In this task i accomplished the following:
- Wrap my App with the Provider component from react-redux
- Passed my configured store into the <Provider>
- Created an initialState variable for my booksSlice
- Displayed my books, received from the slice, in a reusable component
- Dispatched actions using useDispatch

### CODE REVIEW

File: src/components/bookActions.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  const handleAddBook = () => {
+    const newBook = {
+      item_id: 'item4', // Generate a unique ID
+      title: 'New Book',
+      author: 'New Author',
+      category: 'Fiction',
+    };
+    dispatch(addBook(newBook)); // Dispatch the addBook action
+  };
```
Comment (julie-ify) [inline src/components/bookActions.js:16]: - [x] I cannot add a book. I don't think the book addition feature is meant to be coded here. They should be done in the bookForm component. Kindly ensure books can be added.

- [x] The remove book button should be on the book component itself, not beside the add button. That is, each book should have its own remove button.

  image

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+    author: 'Leo Tolstoy',
+    category: 'Fiction',
+  },
+  {
+    item_id: 'item3',
+    title: 'The Selfish Gene',
+    author: 'Richard Dawkins',
+    category: 'Nonfiction',
+  },
+];
```
Comment (julie-ify) [inline src/redux/books/booksSlice.js:23]: - [x] The initialized books should be loaded onto the page when the app is loaded. Kindly ensure this happens.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @mershark,

Good job so far!

## Highlights!
- Store passed in Provider wrapper
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @mershark 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## To highlight :champagne: 
- Nice use of redux :heavy_check_mark: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**